### DISCUSSION TITLE
Will `tbl-pos` be available in YAML header?

### DISCUSSION THREAD

**Discussion by AaronGullickson** [quarto-dev/quarto-cli#8993] [Q&A] [answered]
### Description

I have noticed that you can now add `tbl-pos` as an option to a code chunk to specify the position of table floats for latex output. Its not documented on quarto.org and I am not sure when this was added, but I greatly appreciate the feature! I noticed, however, that I have to specify it on a per-chunk basis. Unlike, `fig-pos`, I cannot specify it as a pdf format option so that it will apply across the whole document, like:

```yaml
format:
    pdf:
       fig-pos: "!t"
       tbl-pos: "!t"
```

I am wondering if that will be a possibility in the future?

**Comment by cscheid** [comment, marked as answer]
I believe this is related to https://github.com/quarto-dev/quarto-cli/issues/7665; we'd need to teach knitr about the generalized syntax for crossref options and make sure it's reflected everywhere in the document.

Quarto's design around Pandoc attributes makes it in general impossible for us to prevent these kinds of unintended features from happening. If we haven't documented a feature, then it's really more of an accident that it works rather than anything else, and we can't promise that it will work in the future.

**Reply by AaronGullickson** [reply]
A happy accident at the moment! Although a little concerning that it could stop working in the future. One of the reasons I ask is that I am working on a PR (rstudio/gt#1588) for the gt package to allow for table floats and it would be nice to know that this argument will work. Right now it works perfectly. How much work is it to teach knitr?

**Reply by wklimowicz** [reply]
Just to tag onto this, I'm currently using the `tbl-pos` syntax in my documents, in order to use the `[H]` placement for raw LaTeX tabulars. I create the tabular from an R script, and I create a cross-reference div with the `tbl-pos` argument to create a table with a caption from it. It would be nice to have confirmation that `tbl-pos` will be supported, or how to make situations like this work in general (LaTeX tabular from file to a table in quarto with a `-pos` argument), although for now `tbl-pos` works fine:

````qmd
---
title: tbl-pos
format: pdf
keep-tex: true
---

With fig-pos="H" nothing happens (it produces `\begin{table}`).

With tbl-pos="H" it produces `\begin{table}[H]` as intended.

::: {#tbl-foo tbl-pos="H"}
```{=latex}
\begin{tabular}{lr}
Foo & Bar \\
Test & Table
\end{tabular}
```
This is a test table caption.
:::
````
